[OSX] Horovod build fails on Tensorflow 2.6.0 on master branch

**Environment:**
Tensorflow 2.6.0
Keras 2.6.0
OSX: 11.5.1

**Bug report:**
Build command:
```
HOROVOD_WITHOUT_MPI=1 HOROVOD_WITHOUT_MXNET=1 HOROVOD_WITHOUT_PYTORCH=1 HOROVOD_WITH_TENSORFLOW=1 python setup.py install
```

Error:

```
[ 71%] Linking CXX shared library ../../../../lib.macosx-11-x86_64-3.7/horovod/tensorflow/mpi_lib.cpython-37m-darwin.so
cd /Users/chongxiaoc/git/horovod/build/temp.macosx-11-x86_64-3.7/RelWithDebInfo/horovod/tensorflow && /usr/local/Cellar/cmake/3.19.0/bin/cmake -E cmake_link_script CMakeFiles/tensorflow.dir/link.txt --verbose=1
/Library/Developer/CommandLineTools/usr/bin/c++ -I/Users/chongxiaoc/build/hvd-py3.7/lib/python3.7/site-packages/tensorflow/include -D_GLIBCXX_USE_CXX11_ABI=0 -DEIGEN_MAX_ALIGN_BYTES=64  -pthread -fPIC -Wall -ftree-vectorize  -O3 -g -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk -dynamiclib -Wl,-headerpad_max_install_names  -undefined dynamic_lookup -Wl,-exported_symbols_list,/Users/chongxiaoc/git/horovod/horovod.exp -o ../../../../lib.macosx-11-x86_64-3.7/horovod/tensorflow/mpi_lib.cpython-37m-darwin.so -install_name @rpath/mpi_lib.cpython-37m-darwin.so CMakeFiles/tensorflow.dir/__/common/common.cc.o CMakeFiles/tensorflow.dir/__/common/controller.cc.o CMakeFiles/tensorflow.dir/__/common/fusion_buffer_manager.cc.o CMakeFiles/tensorflow.dir/__/common/group_table.cc.o CMakeFiles/tensorflow.dir/__/common/half.cc.o CMakeFiles/tensorflow.dir/__/common/logging.cc.o CMakeFiles/tensorflow.dir/__/common/message.cc.o CMakeFiles/tensorflow.dir/__/common/operations.cc.o CMakeFiles/tensorflow.dir/__/common/parameter_manager.cc.o CMakeFiles/tensorflow.dir/__/common/process_set.cc.o CMakeFiles/tensorflow.dir/__/common/response_cache.cc.o CMakeFiles/tensorflow.dir/__/common/stall_inspector.cc.o CMakeFiles/tensorflow.dir/__/common/thread_pool.cc.o CMakeFiles/tensorflow.dir/__/common/timeline.cc.o CMakeFiles/tensorflow.dir/__/common/tensor_queue.cc.o CMakeFiles/tensorflow.dir/__/common/ops/collective_operations.cc.o CMakeFiles/tensorflow.dir/__/common/ops/operation_manager.cc.o CMakeFiles/tensorflow.dir/__/common/optim/bayesian_optimization.cc.o CMakeFiles/tensorflow.dir/__/common/optim/gaussian_process.cc.o CMakeFiles/tensorflow.dir/__/common/utils/env_parser.cc.o CMakeFiles/tensorflow.dir/__/common/gloo/gloo_context.cc.o CMakeFiles/tensorflow.dir/__/common/gloo/gloo_controller.cc.o CMakeFiles/tensorflow.dir/__/common/gloo/http_store.cc.o CMakeFiles/tensorflow.dir/__/common/gloo/memory_store.cc.o CMakeFiles/tensorflow.dir/__/common/ops/gloo_operations.cc.o CMakeFiles/tensorflow.dir/mpi_ops.cc.o CMakeFiles/tensorflow.dir/xla_mpi_ops.cc.o  -L/Users/chongxiaoc/build/hvd-py3.7/lib/python3.7/site-packages/tensorflow -ltensorflow_framework.2 -L/Users/chongxiaoc/build/hvd-py3.7/lib/python3.7/site-packages/tensorflow/python/ -l:_pywrap_tensorflow_internal.so ../../third_party/compatible_gloo/gloo/libcompatible_gloo.a /usr/local/Cellar/libuv/1.41.0/lib/libuv.a -lpthread
ld: library not found for -l:_pywrap_tensorflow_internal.so
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../../lib.macosx-11-x86_64-3.7/horovod/tensorflow/mpi_lib.cpython-37m-darwin.so] Error 1
make[1]: *** [horovod/tensorflow/CMakeFiles/tensorflow.dir/all] Error 2
```

--- a/cmake/Modules/FindTensorflow.cmake
+++ b/cmake/Modules/FindTensorflow.cmake
@@ -21,8 +21,10 @@ if (LEN EQUAL "4")
     list(GET Tensorflow_OUTPUT 2 Tensorflow_LIBRARIES)
     string(REPLACE " " ";" Tensorflow_LIBRARIES_LIST "${Tensorflow_LIBRARIES}")
     list(GET Tensorflow_LIBRARIES_LIST 0 Tensorflow_LIB_PATH)
-    if (Tensorflow_VERSION VERSION_GREATER_EQUAL "2.6")
+    if ((Tensorflow_VERSION VERSION_GREATER_EQUAL "2.6") AND (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin"))
         # XLA implementations are in _pywrap_tensorflow_internal.so
+        # See https://github.com/horovod/horovod/issues/3132
+        # _pywrap_tensorflow_internal.so is not working correctly for linking on OSX
         set(Tensorflow_LIBRARIES "${Tensorflow_LIBRARIES} ${Tensorflow_LIB_PATH}/python/ -l:_pywrap_tensorflow_internal.so")
     endif()
     message("Tensorflow_LIBRARIES := ${Tensorflow_LIBRARIES}")

--- a/horovod/tensorflow/CMakeLists.txt
+++ b/horovod/tensorflow/CMakeLists.txt
@@ -59,7 +59,11 @@ set(Tensorflow_CXX11 ${Tensorflow_CXX11} PARENT_SCOPE)
 
 # TF SOURCES
 list(APPEND TF_SOURCES "${PROJECT_SOURCE_DIR}/horovod/tensorflow/mpi_ops.cc")
-list(APPEND TF_SOURCES "${PROJECT_SOURCE_DIR}/horovod/tensorflow/xla_mpi_ops.cc")
+if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  # See https://github.com/horovod/horovod/issues/3132
+  # Disable XLA for OSX
+  list(APPEND TF_SOURCES "${PROJECT_SOURCE_DIR}/horovod/tensorflow/xla_mpi_ops.cc")
+endif()
 
 # Create library
 set_output_dir()

